### PR TITLE
bugfix/16759-rangeSelector-overwrite-dataGrouping

### LINKED ISSUE #16759: rangeSelector buttons overwrite series dataGrouping options when selected by default
#### Expected behaviour
When rangeSelector buttons do not specify an approximation method ([`rangeSelector.buttons.dataGrouping.approximation`](https://api.highcharts.com/highstock/rangeSelector.buttons.dataGrouping.approximation)) and one button is selected by default ([`rangeSelector.selected`](https://api.highcharts.com/highstock/rangeSelector.selected)) the approximation method defined in each Series should be preserved ([`series.line.dataGrouping.approximation`](https://api.highcharts.com/highstock/series.line.dataGrouping.approximation)).

#### Actual behaviour
The expected behaviour is already present when `rangeSelector.selected` is `undefined`: the button `dataGrouping` options are used to patch the series, so any missing property in the button is preserved in the series' `dataGrouping` options.

However, when a button is selected initially through `rangeSelector.selected`, the button's entire `dataGrouping` object *replaces* the series' instead of patching it. The demo below illustrates this inconsistency.

My use case is to dispaly multiple series with different data grouping options in the same chart. To this end, I need to specify those options in the series' `dataGrouping` options, but this can not work if the range selector buttons override them.

#### Live demo with steps to reproduce
https://jsfiddle.net/tx1582qp/1/

#### Product version
Highstock 9.3.2 (latest)

#### Affected browser(s)
n/a

### PR DESCRIPTION
**Pull request by mfilipiec**
Fixed #16759, range selector buttons overwrote series data grouping options when selected by default.

### CODE REVIEW

File: samples/unit-tests/series/datagrouping/demo.js [hunk 1]
```diff
@@ -1016,4 +1016,35 @@ QUnit.test('The dataGrouping enabling/disabling.', function (assert) {
             the series.${prop} property should be deleted.`
         );
     });
+});
+
+QUnit.test('The dataGrouping with selected range.', function (assert) {
```
Comment (raf18seb) [inline samples/unit-tests/series/datagrouping/demo.js:1021]: Not really a big deal, but you can use ES6 arrow function here

**Review by raf18seb — CHANGES_REQUESTED** [review]
Thanks! Please see comments below.

- [ ] Also, is it necessary to create the whole QUnit.test? Maybe we could reuse an existing test instead.

File: samples/unit-tests/series/datagrouping/demo.js (hunk 1 above)
Comment (raf18seb) [reply samples/unit-tests/series/datagrouping/demo.js]: Not really a big deal again, but if there's only one argument, you don't need parentheses. But you can leave it as it is not :)

**Comment by mfilipiec** [comment]
> * [x]  Also, is it necessary to create the whole QUnit.test (and create the whole chart)? Maybe we could reuse an existing test instead.

@raf18seb I was trying to use one of the existing tests but neither of them matched in this case.

**Review by raf18seb — CHANGES_REQUESTED** [review]
(edit)
I see, it needs to be checked on first chart load (when xAxis is not created yet).

I found that you can reuse this test:
```js
QUnit.test('DataGrouping and update', function (assert) {
    const chart = Highcharts.stockChart('container', {
        rangeSelector: {
            selected: 0,
            buttons: [{
                type: 'all',
                text: 'All',
                dataGrouping: {
                    forced: true,
                    units: [
                        ['millisecond', [3]]
                    ]
                }
            }]
        },
        series: [
            {
                id: 'usdeur',
                dataGrouping: {
                    forced: true,
                    approximation: () => 3
                },
                data: [1, 2, 3]
            }
        ]
    });

    assert.strictEqual(
        chart.series[0].points[0].y,
        3,
        `When the scope is set in the dataGrouping options
        it should workas well as without it (#16759)`
    );

    chart.update(
        {
            series: [
                {
                    id: 'usdeur',
                    data: [3, 2, 1]
                },
                {
                    id: 'eurusd',
                    data: [1, 2, 3]
                }
            ]
        },
        true,
        true
    );

    assert.ok(
        true,
        `Should be no errors when:
            - updating one-to-one with IDs
            - DG enabled
        (#11471)`
    );
});
```

- [ ] Please, change the test name accordingly.

**Review by TorsteinHonsi — APPROVED** [review]
Thanks!

**Review by raf18seb — APPROVED** [review]
Good job, thanks!

**Pull request merged**